crates/sui-core/src/rpc_index.rs

```diff
@@ -160,22 +205,79 @@ impl IndexStoreTables {
         )
     }
 
-    fn needs_to_do_initialization(&self) -> bool {
-        match self.meta.get(&()) {
-            Ok(Some(metadata)) => metadata.version != CURRENT_DB_VERSION,
-            Ok(None) => true,
-            Err(_) => true,
+    /// Lists all Column Families present in the on-disk database
+    fn list_cfs(&self) -> Vec<String> {
+        let path = self.meta.db.path_for_pruning();
+        typed_store::rocks::list_tables(path.to_path_buf()).unwrap()
```

Comment:
this function is getting deprecated in a concurrent PR.
Can we instead drop the whole database(`safe_drop_db` call) and reopen an empty one instead of manual cf manipulation

Reply:
I originally had that, and it turned out to not be "safe". I couldn't do anything to actually get the call to return successfully.